### PR TITLE
FEC-10993 SmartSwitch Enablement

### PR DESCRIPTION
**Pull request by GouravSna**

### CODE REVIEW

File: smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        private val resourceKey = "resource"
+        private val originCodeKey = "originCode"
+        private var errorMessage = "Invalid Response"
+
+        override fun call(): Pair<String, String> {
+            var connection: HttpURLConnection? = null
+            val inputStream: InputStream?
+            var smartSwitchUri: Uri
+            try {
+                smartSwitchUri = Uri.parse(smartSwitchServerUrl)
```
Comment (giladna) [inline smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt]: @GouravSna 

maybe call this  in 1 line?

smartSwitchUri = appendQueryParams(Uri.parse(smartSwitchServerUrl))

File: smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                }
+                return error(resourceUrl, errorMessage)
+            } catch (exception: IOException) {
+                log.d("SmartSwitch IOException: ${exception.message}")
+                exception.message?.let {
+                    errorMessage = "SmartSwitch IOException: $it"
+                }
+                return error(resourceUrl, errorMessage)
+            } finally {
+                connection?.disconnect()
```
Comment (giladna) [inline smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt:133]: @GouravSna 

please call  inputStream.close()  in finally 
I think it is also required

File: smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt (hunk 1 above)
Comment (GouravSna) [reply smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt]: Done

File: smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt (hunk 2 above)
Comment (GouravSna) [reply smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt:133]: Done

**Review by giladna — APPROVED** [review]

**Pull request merged**